packages/attune/tests/smoke/main.rs

```diff
@@ -0,0 +1,465 @@
+                panic!("GPG key generation failed. Make sure GPG is installed and configured.");
+            }
+            
+            println!("✅ GPG key generated successfully");
+            let combined_output = format!("{}{}", 
+                String::from_utf8_lossy(&output.stdout),
+                String::from_utf8_lossy(&output.stderr)
+            );
+            println!("GPG output:\n{combined_output}");
+            
+            // Parse the GPG output to find the key ID from the revocation certificate message
+            for line in combined_output.lines() {
+                if line.contains("revocation certificate stored as") && line.contains(".rev'") {
+                    // Extract the key ID from the filename
+                    if let Some(start) = line.rfind('/') {
+                        if let Some(end) = line.rfind(".rev'") {
+                            let key_id = line[(start + 1)..end].to_string();
+                            println!("✅ Extracted GPG key ID from revocation certificate: {key_id}");
+                            
+                            // Clean up config file
+                            if let Err(e) = fs::remove_file(config_path) {
+                                eprintln!("⚠️  Warning: Could not clean up GPG config file: {e}");
+                            }
+                            
+                            // Set the GPG_KEY_ID environment variable
+                            sh.set_var("GPG_KEY_ID", &key_id);
+                            println!("✅ GPG_KEY_ID environment variable set");
+                            
+                            return key_id;
+                        }
+                    }
+                }
+            }
+            eprintln!("❌ Could not find GPG key ID in revocation certificate output:\n{combined_output}");
+            panic!("GPG key ID not found in generation output");
+        }
+        Err(e) => {
+            eprintln!("❌ GPG key generation failed: {e}");
+            panic!("GPG key generation failed. Make sure GPG is installed and configured.");
+        }
+    };
+}
+
+/// Test that the CLI binary exists and is executable.
+fn test_cli_binary_exists(config: &TestConfig) {
+    let sh = Shell::new().unwrap();
+    set_up_shell_env(&sh, config);
+
+    let cli_path = &config.cli_path;
+    let version_result = cmd!(sh, "{cli_path} --help").run();
+
+    match version_result {
+        Ok(_) => {
+            println!("CLI binary is accessible and executable");
+            println!("✅ Test completed successfully!\n");
+        }
+        Err(e) => {
+            eprintln!("❌ CLI binary test failed: {e}");
+            eprintln!("Make sure the CLI is built and the path is correct.");
+            eprintln!("Current CLI path: {}", config.cli_path);
+            eprintln!(
+                "You can set ATTUNE_CLI_PATH environment variable to specify the correct path."
+            );
```

Comment:
[nit] consider using a single `eprintln!` and `indoc` to emit this all as one operation. It's low impact here, but good habit to be in: `eprintln!` and `println!` write directly to the output device instead of buffering, so it's actually quite expensive to do multiple invocations back to back so it's a good habit to be in.